### PR TITLE
Bump OVSDBTimeout and make it configurable.

### PR DESCRIPTION
**Pull request by npinaeva**
Make OVSDBTimeout for transactions a part of config.Global instead of const. A reasonable value for this timeout depends on the size of the cluster and making it configurable may help solve crashloop problems. It also speeds up testing as timeout may be set to lower value for unit tests.
`types.OVSDBTimeout` is still used in `libovsdb.newClient` for inactivity and reconnectTimeouts.
How much time a transaction takes depends on the size of this transaction, one of the most complicated examples is address set dbIDs sync, which needs to update address set name and all its references (i.e. acls, lrps, qoses matches). I have set batch size for transaction to 50 https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/external_ids_syncer/address_set/address_set_sync.go#L76 considering this is small enough, but a customer hit that timeout. 
Then I considered doing something similar to port group code https://github.com/ovn-org/ovn-kubernetes/blob/425a328cd1729d0dc9ce0a52936d47dee1702b4e/go-controller/pkg/ovn/external_ids_syncer/port_group/port_group_sync.go#L42-L76, but it is way too complicated for address sets with 3 different referencing object types, and looks like an overkill just to avoid a valid transaction timing out.
I am not aware of any reasoning for the existing db timeout = 10s, so I am trying to figure out which value makes sense by default. Considering the timeout is only there to avoid rpc call hanging, which should almost never happen, a couple of minutes sounds like a reasonable upper threshold.
Now getting back to 50 address set batch size (as there is not better reference for what should be supported) I measured locally that updating 50 address sets, each referencing 5000 ACLs takes 41s, with 10000 ACLs it takes 80s.
As I have seen clusters using thousands of ACLs, and namespace selector is quite popular (which is converted to the address sets reference in the ACL.match), I assume it is a good enough example to try to support.
That is how I ended up with default value of 100s, to make sure mentioned transactions succeed, but if the rpc call hangs, we can still recover.

### CODE REVIEW

File: go-controller/pkg/config/config.go
Comment (kyrtapz) [inline go-controller/pkg/config/config.go:38]: Any reason for not doing this skipping the GetDbTxnTimeout func?
```suggestion
var DefaultDBTxnTimeout = 100 * time.Second
```

Comment (npinaeva) [reply go-controller/pkg/config/config.go:38]: Mostly because it is a default value for config flag of type uint

**Comment by kyrtapz** [comment]
This is another issue with the libovsdb timeouts that we should eventually address more broadly.
One idea we discussed with @jcaamano was to exclude the time we spend on decoding json (the majority) from the transaction timeouts. As this would require libovsdb rework (including its [dependencies](https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/vendor/github.com/cenkalti/rpc2/client.go#L272)) I am ok to get this in as a solution we can backport.

/lgtm

File: go-controller/pkg/config/config.go
Comment (kyrtapz) [reply go-controller/pkg/config/config.go:38]: It should be straightforward to use time.Duration instead, we are doing it here:
https://github.com/ovn-org/ovn-kubernetes/blob/c04b0861edbd7e8154296a271b3af5da47c4e646/go-controller/pkg/config/config.go#L345

**Review by kyrtapz — APPROVED** [review]

**Review by trozet — CHANGES_REQUESTED** [review]
I'm OK with making the timeout configurable. I'm not OK with making it 10x the current value without detailed evidence as to why that is required.

Also maybe I missed it but the types.OVSDBTimeout was not removed. Is it still being used in other places?

**Comment by npinaeva** [comment]
> I'm OK with making the timeout configurable. I'm not OK with making it 10x the current value without detailed evidence as to why that is required.
> 
> Also maybe I missed it but the types.OVSDBTimeout was not removed. Is it still being used in other places?

Updated the PR description.

**Review by trozet — APPROVED** [review]
It seems like a legitimate problem where we have to execute a bunch of operations in a txn when there are many references to an object being updated. We talked about potentially dynamically checking how many ops are in a txn and then dynamically setting the timeout on each txn, however @npinaeva points out that each op itself can be very long and take a lot of time. Since there is no valid way to generically estimate how long an op will take we cant do a dynamic implementation.

With IC the txn stays within the node and does not cross the network, so if a transaction were to hang it would be due to something wrong with NBDB. In this case NBDB is messed up anyway so recovering quickly on the OVNk side wont help much. Plus it limits the blast radius to a single node. I'm OK with this change for IC.

Non-IC could be more problematic I suppose if the DBs were not colocated with ovnk master. Will wait for @girishmg feedback before merging.

**Pull request merged**
